ncm-afsclt/src/main/perl/afsclt.pm

```diff
@@ -801,197 +368,113 @@ sub update_afs_cells ( $$ ) {
-	$line =~ /-p\s+tcp.*--dport\s+7002/ ||
-	$line =~ /-p\s+udp.*--dport\s+7003/ ||
-	$line =~ /-p\s+udp.*--dport\s+7004/ ) {
-      $self->debug(2, "skipping line with AFS ports");
-      $found =1;
-      next;
-    }
+=over
 
-    $self->debug(2, "copying existing line");
-    push(@newtables, $line);
+=item Configure()
 
-  }
-  if ($found) {
-    my $newtables = join('',@newtables);
-
-    $self->debug(3, "new $iptables:\n".$newtables);
-    $return = (LC::Check::file($iptables,
-			       "contents"    => $newtables
-			      )
-	      );
-  }
-  return $return;
-}
+Sets ThisCell.
 
+=item Unconfigure()
 
-##########################################################################
-sub Unconfigure_Config {
-##########################################################################
-  my ($self,$config)=@_;
+/etc/sysconfig/afs: reset to defaults
+CellServDB: no action
 
-  if (! open(FD, "<$sysconfig_afs")) {
-    $self->error("Cannot unconfigure: error opening file $sysconfig_afs: $!");
-    return 0;
-  }
-  my @sysconfig = <FD>;
-  close (FD);
-  my @newconfig;
-  my $return = 1;
-  my $found = 0;
-
-  foreach my $line (@sysconfig) {
-    $self->debug(5, "found line $line");
-
-    if ($line =~ /^#.*$compname/ ) {
-      $self->debug(2, "removing NCM comment line");
-      $found =1;
-      next; # just skip
-    } elsif ($line =~ /^OPTIONS=/ ) {
-      $self->debug(2, "resetting OPTIONS");
-      $line = "OPTIONS=\$MEDIUM\n";
-      $found =1;
-    } elsif ($line =~ /^VERBOSE=/ ) {
-      $self->debug(2, "resetting VERBOSE");
-      $line = "VERBOSE=\n";
-      $found =1;
-    } elsif ($line =~ /^AFS_CLIENT=/ ) {
-      $self->debug(2, "resetting AFS_CLIENT enable");
-      $line = "AFS_CLIENT=on\n";
-      $found =1;
-    } else {
-      $self->debug(2, "copying existing line");
-    }
-    push(@newconfig, $line);
-  }
+=back
 
-  if ($found) {
-    my $newconfig = join('',@newconfig);
+=head1 RESOURCES
 
-    $self->debug(3, "new $sysconfig_afs:\n".$newconfig);
-    $return = (LC::Check::file($sysconfig_afs,
-			       "contents"    => $newconfig
-			      )
-	      );
-  }
-  return $return;
-}
+=over
 
-##########################################################################
-sub Unconfigure_Cache {
-##########################################################################
-  my ($self,$config)=@_;
-  $self->info("AFS cache config not changed");
-  return 1;
-}
+=item /software/components/afsclt/active : boolean
 
-##########################################################################
-sub Unconfigure_PAM {
-##########################################################################
-  my ($self,$config)=@_;
-  my $libpam;
-  my $libpam_options_auth;
-  my $libpam_options_refresh;
-  my $ret = 0;
+activates/deactivates the component.
 
-  if ($OSname =~ /Solaris/) {
-    $self->warn("Somebody needs to write something to unconfigure PAM on Solaris"); #FIXME
-    return 0;
+=item /software/components/afsclt/thiscell : string
```

Comment:
documentation of the schema is best done via annotations in the schema. so this could be
```pan
type ... {
    ...
    @{local AFS cell for this machine}
     "thiscell" ? string
```
put the `@{}` block before the attribute